src/main/java/com/ministore/api/controllers/CreateUserHandler.java

```diff
@@ -32,7 +33,15 @@ private User createUser(@RequestBody CreateUserRequest request)
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "username is invalid");
         }
 
-        return this.userService.createUser(request.username);
+        User user = null;
+
+        try {
+            user = this.userService.createUser(request.username);
+        } catch(DataIntegrityViolationException exception) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "username is invalid");
```

Comment:
Some comments:
- Http Status Code 409 maybe is a better option to this scenario